hibernate-core/src/main/java/org/hibernate/metamodel/model/domain/ManagedDomainType.java

```diff
@@ -6,10 +6,18 @@
  */
 package org.hibernate.metamodel.model.domain;
 
+import javax.persistence.metamodel.Attribute;
 import javax.persistence.metamodel.ManagedType;
 
 /**
  * @author Steve Ebersole
  */
 public interface ManagedDomainType<J> extends SimpleDomainType<J>, ManagedType<J> {
+	/**
+     *  Return the attribute of the managed
+     *  type that corresponds to the specified name if exists.
+     *  @param name  the name of the represented attribute
+     *  @return attribute if exists else null
+     */
+	Attribute<? super J, ?> getAttributeIfExists(String name); 
```

Comment:
Not sure if a metamodel extension is the way to go here. Especially not adding an alternative to a method already present with just slightly different handling of absent values (not to mention that that method comes from the JPA spec and not just our metamodel).

There must be an other way to prevent absent attributes being looked up...

Reply:
Agreed. Hibernate should not apply entity graph on non root entities in first place.

Reply:
The method itself is a good idea.  The name is awful ;)

Also, there are a lot of extra unnecessary changes here.  I am about to push an alternative PR